File: tests/fixtures/lang_samples/python/complex.py
def many_branches(a, b, c, d, e):
    if a:
        return 1
    elif b:
        return 2
    elif c:
        return 3
    elif d:
        return 4
    elif e:
        return 5
    if a and b:
        return 6
    if c or d:
        return 7
    if a and c and e:
        return 8
    for i in range(10):
        if i:
            continue
    return 0

File: tests/fixtures/lang_samples/pascal/complex.pas
unit Complex;

interface

implementation

function ManyBranches(A, B, C, D, E: Boolean): Integer;
begin
  if A then
  begin
    Result := 1;
  end
  else if B then
  begin
    Result := 2;
  end
  else if C then
  begin
    Result := 3;
  end
  else if D then
  begin
    Result := 4;
  end
  else if E then
  begin
    Result := 5;
  end;

  if A and B then
  begin
    Result := 6;
  end;

  if C or D then
  begin
    Result := 7;
  end;

  if A and C and E then
  begin
    Result := 8;
  end;

  Result := 0;
end;

end.
